src/utils.ts

```diff
@@ -0,0 +1,28 @@
+import * as temp from 'temp'
+
+export function sanitizedDateString(date?: Date) {
+    const d = date || new Date();
+    const pad = (num: number) => ("00" + num).slice(-2);
+
+    return `${d.getFullYear()}-${pad(d.getMonth())}-${pad(d.getDay())}-${pad(d.getHours())}-${pad(d.getMinutes())}-${pad(d.getSeconds())}`;
+}
+
+const tempDirs: { [sharedKey: string]: string } = {};
+export function getSharedTempDir(sharedKey: string): Promise<string> {
+    if (tempDirs[sharedKey]) {
+        return Promise.resolve(tempDirs[sharedKey]);
+    }
+
+    return new Promise((resolve, reject) => {
+        temp.track();
```

Comment:
Earlier I confirmed that it cleared the temp file properly when closing the window or reloading the session/switching folders.

> It is possible to unload and reload the extension without the process actually exiting.

The main VSCode process doesn't exit, but the extension host does. Extensions are run in an independent sandboxed process which is restarted when you reload the window.